core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java

```diff
@@ -355,19 +397,18 @@ private void runInternal() {
                     mContext.getRequest(), e);
                 setError(new Error(AlluxioStatusException.fromThrowable(e), true));
               } finally {
-                if (finalChunk != null) {
-                  finalChunk.release();
-                }
+                finalChunk.release();
               }
             });
           }
         } catch (Exception e) {
           LogUtils.warnWithException(LOG,
-              "Exception occurred while reading data for read request {}.", mContext.getRequest(),
+              "Exception occurred while reading data for read request {}. session {}",
+              mContext.getRequest(), mContext.getRequest().getSessionId(),
               e);
           setError(new Error(AlluxioStatusException.fromThrowable(e), true));
-          continue;
         }
+        continue;
```

Comment:
@ZacBlanco Is there a reason for moving this continue out? Is it possible for this to cause a tight-loop here?

Reply:
While this looks like a typo I made during the PR. Looking at the logic I don't think it should have any effect.

The bracket `}` below the `continue` matches to the outer-loop `while (true) {`. The bracket above matches to the exception handler

Previously, the continue statement was within the exception handler which would have sent the program back to the beginning of the loop.

However, now that `continue` is the last statement in the loop which is essentially a no-op since it would have returned to the beginning anyway. I think even before it would have been a no-op because after exiting the exception handler the program flow would have sent it to the beginning of the loop anyway.